src/cmds/versions/create.js

```diff
@@ -2,104 +2,104 @@ const config = require('config');
+        type: String,
+        description: 'Would you like to make this version public? Any primary version must be public.',
+      },
+    ];
   }
 
-  if (!version || !semver.valid(semver.coerce(version))) {
-    return Promise.reject(
-      new Error(`Please specify a semantic version. See \`${config.get('cli')} help ${exports.command}\` for help.`)
-    );
-  }
+  async run(opts) {
+    let versionList;
+    const { key, version, codename, fork, main, beta, isPublic } = opts;
 
-  if (!fork) {
-    versionList = await fetch(`${config.get('host')}/api/v1/version`, {
-      method: 'get',
-      headers: cleanHeaders(key),
-    }).then(res => handleRes(res));
-  }
+    if (!key) {
+      return Promise.reject(new Error('No project API key provided. Please use `--key`.'));
+    }
 
-  const versionPrompt = promptOpts.createVersionPrompt(versionList || [{}], {
-    newVersion: version,
-    ...opts,
-  });
+    if (!version || !semver.valid(semver.coerce(version))) {
+      return Promise.reject(
+        new Error(`Please specify a semantic version. See \`${config.get('cli')} help ${this.command}\` for help.`)
+      );
+    }
 
-  const promptResponse = await prompt(versionPrompt);
+    if (!fork) {
+      versionList = await fetch(`${config.get('host')}/api/v1/version`, {
+        method: 'get',
+        headers: cleanHeaders(key),
+      }).then(res => handleRes(res));
+    }
 
-  return fetch(`${config.get('host')}/api/v1/version`, {
-    method: 'post',
-    headers: cleanHeaders(key, {
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-    }),
-    body: JSON.stringify({
-      version,
-      codename: codename || '',
-      is_stable: main === 'true' || promptResponse.is_stable,
-      is_beta: beta === 'true' || promptResponse.is_beta,
-      from: fork || promptResponse.from,
-      is_hidden: promptResponse.is_stable ? false : !(isPublic === 'true' || promptResponse.is_hidden),
-    }),
-  })
-    .then(res => res.json())
-    .then(res => {
-      if (res.error) {
-        return Promise.reject(new APIError(res));
-      }
-      return Promise.resolve(`Version ${version} created successfully.`);
+    const versionPrompt = promptOpts.createVersionPrompt(versionList || [{}], {
+      newVersion: version,
+      ...opts,
     });
+
+    const promptResponse = await prompt(versionPrompt);
+
+    return fetch(`${config.get('host')}/api/v1/version`, {
+      method: 'post',
+      headers: cleanHeaders(key, {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      }),
+      body: JSON.stringify({
+        version,
+        codename: codename || '',
+        is_stable: main === 'true' || promptResponse.is_stable,
+        is_beta: beta === 'true' || promptResponse.is_beta,
+        from: fork || promptResponse.from,
+        is_hidden: promptResponse.is_stable ? false : !(isPublic === 'true' || promptResponse.is_hidden),
+      }),
+    })
+      .then(handleRes)
```

Comment:
Moved this over to use `handleRes` because we can here.

Reply:
niiiice